docs/SUMMARY.md

```diff
@@ -0,0 +1,35 @@
+# Summary
+
+- [Introduction](./introduction.md)
+- [Tutorial](./tutorial/index.md)
+  - [Creating `hello-world.wasm`](./tutorial/create-hello-world.md)
+  - [Running `hello-world.wasm`](./tutorial/run-hello-world.md)
+- [Examples](./examples/index.md)
+  - [Markdown parser](./examples/markdown.md)
```

Comment:
One thing that has been kind of a pain to manage for `mdbook`s that I've dealt with in the past is keeping URLs stable. Inevitably, we will want to tweak the hierarchy of this table of contents, but if we also tweak the directory and `md` file structure to match, we break existing links. On top of that, `mdbook` doesn't have a system for creating redirects.

Given that, I think it makes sense to put everything in a top-level directory with slightly more descriptive than normal filenames. This should be a little more robust to re-organizations and hierarchy changes, with less link breakage.

Concretely, I guess I would suggest `./examples/markdown.md` => `./markdown-example.md` or something like that. I understand this might be a bit controversial, so if everyone else feels strongly that this isn't worth it, I'm happy to defer to the group.

Reply:
> Concretely, I guess I would suggest `./examples/markdown.md` => `./markdown-example.md` or something like that.

Can you say more about why that'd be more robust to re-organizations? ISTM we'd want to rename those files in all the same situations in which we'd otherwise want to move files to other directories, but I'm probably missing scenarios.

Reply:
Say we wanted to reorganize the table of contents to move `examples` under a new `reference` section. If we wanted to keep the directory hierarchy matching the table of contents, we would move the `examples` directory into a new `reference` directory too, breaking all the old links to the markdown example or any other example.

With the single flat directory of files, we wouldn't rename the file when changing hierarchies of the table of contents, so the links would continue to work. True, the markdown example file wouldn't have "reference" anywhere in its name now, but the links would still work.

Alternatively, we could leave the file at `./example/markdown.md` even when the table of contents has moved to the hierarchy `References >> Examples >> Markdown`, but now there are two different nestings to keep track of (one for the table of contents and one for the directory structure and URLs) and it seems like unnecessary mental overhead to have both.

Reply:
I'd be fine either way personally, this is largely just an organization thing on our end so I think it's fine to do whatever we feel is best